### PR TITLE
bugfix: S3C-3350 avoid throwing exception for legacy objects

### PR DESCRIPTION
**Pull request by rahulreddy**
bugfix: S3C-3350 avoid throwing exception for legacy objects 

Note: I don't have test setup as it's a compatibility test. I plan on following up on a test for this later post cc.

### CODE REVIEW

File: lib/api/apiUtils/object/partInfo.js [hunk 1]
```diff
@@ -20,13 +20,23 @@ function getPartNumber(query) {
...
 function getPartSize(objMD, partNumber) {
     let size;
+    let locationPartNumber;
     if (partNumber && objMD && objMD.location
         && objMD.location.length >= partNumber) {
         const locations = [];
         for (let i = 0; i < objMD.location.length; i++) {
             const { dataStoreETag } = objMD.location[i];
-            const locationPartNumber =
+            if (!dataStoreETag) {
```
Comment (naren-rajendran) [inline lib/api/apiUtils/object/partInfo.js]: May be reverse the condition, `if (dataStoreETag) {} else {}` would look cleaner.

File: lib/api/objectGet.js [hunk 2]
```diff
@@ -156,17 +156,21 @@ function objectGet(authInfo, request, returnTagCount, log, callback) {
...
-                            .customizeDescription('PartNumber parameter for ' +
-                            'this object is not supported'));
-                    }
-                    const locationPartNumber =
+                    /**
+                     * Location objects prior to GA7.1 do not include the
+                     * dataStoreETag field so we cannot find the part range, the
+                     * objects are treated as if they only have 1 part
+                     */
+                    if (!dataStoreETag) {
```
Comment (naren-rajendran) [inline lib/api/objectGet.js]: Same.

**Comment by naren-rajendran** [comment]
Looks good to me, just recommending a minor change.

**Review by naren-rajendran — APPROVED** [review]

**Review by ilkescality — APPROVED** [review]

File: lib/api/apiUtils/object/partInfo.js (hunk 1 above)
Comment (rahulreddy) [reply lib/api/apiUtils/object/partInfo.js]: Agreed

**Pull request merged**